typings/audit.d.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+declare namespace LH.Audit {
+  export interface ScoringModes {
```

Comment:
enum?

Reply:
it looks like we can't (at least in a `d.ts` file). Since `enum` defines a type and a value in ts, it expects you to use the value of the enum anywhere the type is expected. So e.g. the values of `Audit.SCORING_MODES` would have to be `LH.Audit.ScoringModes` (not just 'numeric' and 'binary', even though that's what those are), but those don't exist in JS if the enum is defined here.

Reply:
I could be wrong, but I can't make it work. [Playground example](https://www.typescriptlang.org/play/#src=enum%20ScoringMode%20%7B%0D%0A%20%20%20%20NUMERIC%20%3D%20'numeric'%2C%0D%0A%20%20%20%20BINARY%20%3D%20'binary'%0D%0A%7D%0D%0A%0D%0Atype%20ScoringModeObj%20%3D%20%7B%0D%0A%20%20%20%20%5BK%20in%20keyof%20typeof%20ScoringMode%5D%3A%20typeof%20ScoringMode%5BK%5D%0D%0A%7D%0D%0A%0D%0Aclass%20Audit%20%7B%0D%0A%20%20%20%20static%20get%20SCORING_MODES()%3A%20ScoringModeObj%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20NUMERIC%3A%20'numeric'%2C%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20BINARY%3A%20'binary'%2C%0D%0A%20%20%20%20%20%20%20%20%7D%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)

Reply:
you mean you're not a fan of [this](https://www.typescriptlang.org/play/#src=enum%20ScoringMode%20%7B%0D%0A%20%20%20%20NUMERIC%20%3D%20'numeric'%2C%0D%0A%20%20%20%20BINARY%20%3D%20'binary'%0D%0A%7D%0D%0A%0D%0Atype%20ScoringModeObj%20%3D%20%7B%0D%0A%20%20%20%20%5BK%20in%20keyof%20typeof%20ScoringMode%5D%3A%20ScoringMode%0D%0A%7D%0D%0A%0D%0A%2F**%0D%0A%20*%20%40param%20%7Bany%7D%20s%0D%0A%20*%20%40return%20%7BScoringMode%7D%0D%0A%20*%2F%0D%0Afunction%20castToScoringMode(s)%20%7B%0D%0A%20%20%20%20return%20s%0D%0A%7D%0D%0A%0D%0Aclass%20Audit%20%7B%0D%0A%20%20%20%20static%20get%20SCORING_MODES()%3A%20ScoringModeObj%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20NUMERIC%3A%20castToScoringMode('numeric')%2C%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20BINARY%3A%20castToScoringMode('binary')%2C%0D%0A%20%20%20%20%20%20%20%20%7D%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)!?

😆